Source/Analytics/Web/src/components/Map.js

```diff
@@ -3,43 +3,113 @@ import { connect } from "react-redux";
 import { QueryCoordinator } from "@dolittle/queries";
 
-var redCrossIcon = L.icon({
-    iconUrl: 'https://upload.wikimedia.org/wikipedia/commons/thumb/1/1a/Flag_of_the_Red_Cross.svg/250px-Flag_of_the_Red_Cross.svg.png',
-    iconSize: [10, 10], // size of the icon
-});
+var clusterSize = {
+    large: 40,
+    medium: 30,
+    small: 20
+};
+
+var clusters     = []
+var createClusterCustomIcon = function(cluster) {
+    var markersInCluster  = cluster.getChildCount();
+
+    // Get css class from markers
+    var getClusterNbr = cluster.getAllChildMarkers()[0].options.icon.options.className.split(" ")[1]
 
+    var markers = cluster.getAllChildMarkers();
 
-function CaseMarkers({casesLastWeekAndMonth}){
-    let id = (Object.keys(casesLastWeekAndMonth.caseReportsPerHealthRisk)[0]);
-    var casesForOneHealthRisk = casesLastWeekAndMonth.caseReportsPerHealthRisk[id];
-    console.log(casesForOneHealthRisk.healthRiskName);
+    var markersSet = {}
+
+    for(var i=0; i<markers.length; i++){
+        var tmp_mark = markers[i].options.icon.options.className.split(" ")[1]
+        if(!(tmp_mark in markersSet)){
+            markersSet[tmp_mark] = 0
+        }else{
+            markersSet[tmp_mark]++
+        }
+    }
 
-    Object.keys(casesLastWeekAndMonth.caseReportsPerHealthRisk).forEach(function(key) {
-        console.log(casesLastWeekAndMonth.caseReportsPerHealthRisk[key].caseReportsLast7Days[0]);
-     });
+    // Get healthRiskNumber from css class 
+    var healthRiskNbr = getClusterNbr[getClusterNbr.length - 1]
+    var markerClusterSize
 
-    return casesForOneHealthRisk.caseReportsLast7Days.map(cases => {
-        cases.vals = new Array(cases.numberOfPeople);
-        cases.vals.fill(1);
-        return cases.vals.map(function(val){  
-            return <Marker
-                position={[cases.location.longitude, cases.location.latitude]} icon={redCrossIcon}
-            ></Marker>
+    if(markersInCluster <= 5){
+        markerClusterSize = clusterSize.small
+    }
+    else if(markersInCluster < 20){
+        markerClusterSize = clusterSize.medium
+    }
+    else{
+        markerClusterSize = clusterSize.large
+    }
+
+
+    if(Object.keys(markersSet).length > 1){
+        healthRiskNbr = 5
+    }
+    return new L.divIcon({
+        html: `${cluster.getChildCount()}`,
+        className: "marker-" + healthRiskNbr + "-cluster marker-cluster-custom",
+        iconSize: L.point(markerClusterSize, markerClusterSize, true )
         });
-    });   
+
+  }
+
+
+
+var healthRisks  = []
+var iconsClasses = []
+
+function MarkerPopupContent(props){
+    return <div>{props.healthRisk}</div>
+}
+
+function CaseMarkers({caseReportsLastWeek}){
+    // Returns one Marker for each case in a case-report and sets an unique color for each specific health risk
+    return  Object.keys(caseReportsLastWeek.caseReportsPerHealthRisk).map(function(key) {
+        var healthRiskName   = caseReportsLastWeek.caseReportsPerHealthRisk[key][0].healthRiskName
```

Comment:
number of spaces before =

Reply:
This should also handle the case of no data (caseReportsLastWeek = undefined) so that it doesn't throw an error.